### ISSUE TITLE
Prompt contains garbage characters

### ISSUE THREAD

**Issue by greencardamom** [fish-shell/fish-shell#10160]
Fish version 3.6.4
Linux Mint 19.3 Tricia
echo $TERM is vt100
Linux acre 5.4.0-150-generic #167~18.04.1-Ubuntu SMP Wed May 24 00:51:42 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

GNOME Terminal 3.28.1
Terminal Compatibility Settings: 
Encoding -> Unicode UTF8
Backspace key generates -> ASCII DEL
Delete key generates -> Escape Sequence
Ambiguous width character -> Narrow

I tried changing some of the compat settings, but there are so many permutations it's hard to guess. 

Example exactly what it looks like:

acre:[/home/greenc/scripts] fish
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
acre:[/home/greenc/scripts] 
⏎$<2>
$<3>acre:[/home/greenc/scripts] 

--------------

Tried this:

acre:[/home/greenc/scripsh -c 'env HOME=$(mktemp -d) XDG_CONFIG_HOME= fish'
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
greenc@acre /h/g/scripts> 
⏎$<2>
$<3>greenc@acre /h/g/scripts> 

--------------

I made a custom prompt 

$<3>acre:[/home/greenc/.config/fish/functiocat fish_prompt.fish 
function fish_prompt -d "Write out the prompt"
    printf '%s:[%s] ' $hostname (pwd)
end
⏎$<2>
$<3>acre:[/home/greenc/.config/fish/functions] 

But deleting doesn't change the problem.

**Comment by faho** [comment]
> echo $TERM is vt100

That's not normal. "vt100" describes a 1970s physical terminal. That's not what you are running.

You want the default, which for Gnome Terminal typically is xterm-256color. If you changed the setting, change it back. If you didn't, figure out what broke it.

That's why fish acts in a way that's incompatible with your actual terminal.

**Comment by greencardamom** [comment]
Worked!

acre:[/home/greenc] echo $TERM
xterm-256color
acre:[/home/greenc] fish
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
acre:[/home/greenc] echo $TERM
xterm-256color
acre:[/home/greenc] 

----

I'm using a tcsh .login file that was probably created in the 1980s or maybe earlier (I got it from a co worker in the 80s) and then I copied it from computer to computer over the years, first time this became a problem. Thank you very much.
